File: receipt-analyzer-service/adapters/src/main/java/pl/madzierski/daniel/user/UserController.java
package pl.madzierski.daniel.user;

import lombok.AllArgsConstructor;
import org.springframework.http.MediaType;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.security.oauth2.jwt.Jwt;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@AllArgsConstructor
@RestController
@RequestMapping(path = "/users", produces = MediaType.APPLICATION_JSON_VALUE)
@Validated
public class UserController {

    private final UserFacade userFacade;

    @PostMapping("/sync/me")
    void syncMe(@AuthenticationPrincipal Jwt jwt) {
        userFacade.syncMe(jwt.getSubject());
    }
}

File: receipt-analyzer-service/src/main/kotlin/pl/madzierski/daniel/UserController.kt
package pl.madzierski.daniel

import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import pl.madzierski.daniel.security.SecurityUtils


@Controller
@RestController
@RequestMapping("/users")
class UserController {

    @GetMapping("/me")
    fun me(): ResponseEntity<UserInfo> {
        return ResponseEntity.ok(UserInfo(SecurityUtils.getCurrentUserSub(), SecurityUtils.getCurrentUserRoles()))
    }
}

class UserInfo(val username: String, val all: List<String>)
